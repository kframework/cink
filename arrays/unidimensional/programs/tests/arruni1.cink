void main() {
  int a[5];
  a[0] = 1;
  a[3] = a[0];
  cout << "a[5] = ";
  cin >> a[5];
  cout << a[0] << ";" << a[3] << ";" << a[5];
}

