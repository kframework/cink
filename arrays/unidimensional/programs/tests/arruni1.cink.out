a[5] = 1;1;3<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <store>
        seq-array ( 5 , #symLoc(2) ) |-> store ( store ( store ( 
          const-array ( 5 , 0 ) , 0 , 1 ) , 3 , 1 ) , 5 , 3 )
        #symScalLoc(1) |-> lambda (  @ int (a [ 5 ]) ; ((a [ 0 ]) = 1) ; ((a [
           3 ]) = (a [ 0 ])) ; (cout << "a[5] = ") ; (cin >> (a [ 5 ])) ; ((((
          (cout << (a [ 0 ])) << ";") << (a [ 3 ])) << ";") << (a [ 5 ])) ; )
          
    </store>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
</T>
